test/simulator/timestamp/simulator_interface/simulator_interface.cpp

```diff
@@ -76,19 +77,116 @@ choose_num(int min, int max, const std::string &cli_str)
     return (choice);
 }
 
+const std::string
+get_input(const std::string &input_name)
+{
+    std::string input;
+    std::cout << input_name;
+    std::getline(std::cin, input);
+    std::cout << std::endl;
+    return (input);
+}
+
 void
 interface_session_management()
 {
 }
 
 void
-interface_set_timestamp()
+interface_set_timestamp(connection_simulator *conn)
 {
+    std::string config;
+    std::vector<std::string> options;
+    options.push_back("Oldest timestamp");
+    options.push_back("Stable timestamp");
+    options.push_back("All durable timestamp");
+    options.push_back("<- go back");
+
+    do {
+        try {
+            print_options(options);
+
+            int choice = choose_num(1, options.size(), "Set timestamp >>");
+
+            switch (choice) {
+            case 1:
+                config = "oldest_timestamp=";
+                break;
+            case 2:
+                config = "stable_timestamp=";
+                break;
+            case 3:
+                config = "durable_timestamp=";
+                break;
+            case 4:
+            default:
+                return;
+            }
+
+            config += get_input("Enter timestamp (in hex): ");
+
+            int ret = conn->set_timestamp(config);
+
+            if (ret != 0)
+                throw "'set_timestamp' failed with ret value: '" + std::to_string(ret) +
+                  "', and config: '" + config + "'";
+
+            print_border_msg("- Global timestamp set - " + config, GREEN);
+        } catch (const std::string &exception_str) {
+            print_border_msg("exception: " + exception_str, RED);
+        }
+
+    } while (true);
 }
 
 void
-interface_conn_query_timestamp()
+interface_conn_query_timestamp(connection_simulator *conn)
 {
+    std::string config;
+    std::vector<std::string> options;
+    options.push_back("Oldest timestamp");
+    options.push_back("Stable timestamp");
+    options.push_back("All durable timestamp");
+    options.push_back("<- go back");
+
+    do {
+        try {
+            print_options(options);
+
+            int choice = choose_num(1, options.size(), "Query timestamp >>");
+
+            switch (choice) {
+            case 1:
+                config = "get=oldest";
+                break;
+            case 2:
+                config = "get=stable";
+                break;
+            case 3:
+                config = "get=all_durable";
+                break;
+            case 4:
+            default:
+                return;
+            }
+
+            std::string hex_ts;
+            bool ts_supported;
+            int ret = conn->query_timestamp(config, hex_ts, ts_supported);
+
+            if (hex_ts == "0")
+                hex_ts = "not specified";
+
+            if (ret != 0)
```

Comment:
Should we check this before checking the output `hex_ts`?

Reply:
Thanks, updated in [f78f929](https://github.com/wiredtiger/wiredtiger/pull/8417/commits/f78f929a04e0d3cc79069fffd5a18b52f823476f)